lib/Conversation/index.js

```diff
@@ -4,18 +4,16 @@ import pluralize from 'pluralize';
 import CommentList from './CommentList';
 import CommentForm from './CommentForm';
 import Button from '../Button';
-import BoundaryClickWatcher from '../BoundaryClickWatcher';
 import ResolveIcon from '../../assets/icons/resolve.svg';
 
 class Conversation extends Component {
   static propTypes = {
     id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
-    actions: PropTypes.shape({
-      resolveConversation: PropTypes.func.isRequired,
-      addComment: PropTypes.func.isRequired,
-      activateConversation: PropTypes.func.isRequired,
-      deactivateConversation: PropTypes.func.isRequired
-    }).isRequired,
+    resolveConversation: PropTypes.func,
+    onCancel: PropTypes.func,
+    addComment: PropTypes.func.isRequired,
+    activateConversation: PropTypes.func,
```

Comment:
Lets abstract out `activateConversation` and `deactivateConversation` and treat BoundaryClickWatcher as another UI component that can be used to compose the interactions we desire.